app/lib/features/chat/pages/edit_chat_page.dart

```diff
@@ -0,0 +1,218 @@
+      dialogTitle: L10n.of(context).uploadAvatar,
+      type: FileType.image,
+    );
+    if (result != null && result.files.isNotEmpty) {
+      selectedUserAvatar.value = File(result.files.first.path!);
+    }
+  }
+
+  Widget _buildAvatarUI() {
+    return ValueListenableBuilder(
+      valueListenable: selectedUserAvatar,
+      builder: (context, userAvatar, child) {
+        return GestureDetector(
+          onTap: () => pickAvtar(),
+          child: Container(
+            height: 100,
+            width: 100,
+            decoration: BoxDecoration(
+              image: userAvatar != null
+                  ? DecorationImage(
+                      image: FileImage(userAvatar),
+                      fit: BoxFit.cover,
+                    )
+                  : null,
+              color: Theme.of(context).colorScheme.primaryContainer,
+              borderRadius: BorderRadius.circular(5),
+            ),
+            child: userAvatar == null
+                ? Icon(
+                    Atlas.up_arrow_from_bracket_thin,
+                    color: Theme.of(context).colorScheme.neutral4,
+                  )
+                : null,
+          ),
+        );
+      },
+    );
+  }
+
+  Widget _chatName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        Padding(
+          padding: const EdgeInsets.only(bottom: 5),
+          child: Text(L10n.of(context).chatName),
+        ),
+        InputTextField(
+          hintText: L10n.of(context).typeName,
+          textInputType: TextInputType.multiline,
+          controller: chatNameController,
+        ),
+        const SizedBox(height: 3),
+        Text(
+          L10n.of(context).egGlobalMovement,
+          style: Theme.of(context).textTheme.labelSmall!.copyWith(
+                color: Theme.of(context).colorScheme.neutral4,
+              ),
+        ),
+      ],
+    );
+  }
+
+  Widget _chatDescription() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        Padding(
+          padding: const EdgeInsets.only(bottom: 5),
+          child: Text(L10n.of(context).about),
+        ),
+        InputTextField(
+          hintText: L10n.of(context).description,
+          textInputType: TextInputType.multiline,
+          controller: chatDescriptionController,
+          maxLines: 10,
+        ),
+      ],
+    );
+  }
+
+  Future<void> _save() async {
+    if (chatNameController.text.trim().isEmpty) return;
+    final convo = await ref.read(chatProvider(widget.roomId).future);
+    if (!mounted) return;
+    EasyLoading.show(status: L10n.of(context).updatingChat);
+    try {
+      await convo.setName(chatNameController.text.trim());
+      await convo.setTopic(chatDescriptionController.text.trim());
+      if (selectedUserAvatar.value != null) {
+        await convo.uploadAvatar(selectedUserAvatar.value!.path);
+      }
+      EasyLoading.dismiss();
```

Comment:
each one of them is a separate matrix room event, if they didn't change, we should be sending the event as that just produces noise ...